tests/data_processing/test_get_flight_stats.py

```diff
@@ -0,0 +1,151 @@
+    ]
+
+    def assertDataframeEqual(self, a: pd.DataFrame, b: pd.DataFrame, msg: None = None) -> None:
+        try:
+            pd_testing.assert_frame_equal(a, b)
+        except AssertionError as e:
+            raise self.failureException(msg) from e
+
+    def setUp(self) -> None:
+        self.addTypeEqualityFunc(pd.DataFrame, self.assertDataframeEqual)
+
+    def test_count_flights_structure(self) -> None:
+        self.data_path = 'data/tests/model_test_subset_dataset.json'
+        with open(self.data_path, 'r', encoding='utf-8') as f:
+            self.raw_data = json.load(f)
+        self.airport_data = DataConverter(self.raw_data)
+        airport_df = self.airport_data.airports_df
+        flights_df = self.airport_data.flights
+        flights_df = pd.concat([flights_df, airport_df['iata']], axis=1, join='outer')
+
+        flights_stats = FlightStats(flights_df)
+        # check output dataframe types are int
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_arrivals']))
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_departures']))
+        # check output shape
+        self.assertEqual(flights_stats.flight_stats_df.shape[1], len(self.FLIGHT_COUNT))
+        # check column names are as expected
+        self.assertEqual(set(flights_stats.flight_stats_df.columns), set(self.FLIGHT_COUNT))
+        # check output is a dataframe
+        self.assertIsInstance(flights_stats.flight_stats_df, pd.DataFrame)
+
+    def test_count_flights_logic_base_case(self) -> None:
+        # base test case
+        fake_input_flights_df = pd.DataFrame(
+            {
+                'iata': 'LBG',
+                'flights': [
+                    [
+                        {'origin_iata': 'LBG', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'LBG', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'LBG', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'SAY', 'destination_iata': 'LBG'},
+                        {'origin_iata': 'YVR', 'destination_iata': 'LBG'}
+                    ]
+                ]
+            },
+            {
+                'iata': 'YVR',
+                'flights': [
+                    [
+                        {'origin_iata': 'YVR', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'YVR', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'YVR', 'destination_iata': 'YYZ'},
+                        {'origin_iata': 'SAY', 'destination_iata': 'YVR'},
+                        {'origin_iata': 'YVR', 'destination_iata': 'YVR'}
+                    ]
+                ]
+            }
+        )
+        flights_stats = FlightStats(fake_input_flights_df)
+
+        fake_output_flights_df = pd.DataFrame(
+            {
+            'total_arrivals': [2, 2],
+            'total_departures': [3, 3]
+            }
```

Comment:
I'm a bit confused by the structure here. How do you know what airport `total_arrivals[0]` corresponds to? Is it `YVR` or `LBG`? Would it not make more sense to have each airport as a row with the `IATA` as the index and arrivals/departures as integers?

Reply:
I guess it could be done yeah - its just based off of order of input right now. It get's merged with the rest of statistics at the end of preprocessing in model.py based off the shared order then its associated again with it's iata.  I guess it would be safer to keep iata linked to each stat.